test/simulator/timestamp/simulator_interface/simulator_interface.cpp

```diff
@@ -0,0 +1,645 @@
+                throw "'set_timestamp' failed with ret value: '" + std::to_string(ret) +
+                  "', and config: '" + config + "'";
+
+            print_border_msg("- Global timestamp set - " + config, GREEN);
+        } catch (const std::string &exception_str) {
+            print_border_msg("exception: " + exception_str, RED);
+        }
+
+    } while (true);
+}
+
+void
+interface_conn_query_timestamp(connection_simulator *conn)
+{
+    std::string config;
+    std::vector<std::string> options;
+    options.push_back("Oldest timestamp");
+    options.push_back("Stable timestamp");
+    options.push_back("All durable timestamp");
+    options.push_back("<- go back");
+
+    do {
+        try {
+            print_options(options);
+
+            int choice = choose_num(1, options.size(), "[Conn] query timestamp >>");
+
+            switch (choice) {
+            case 1:
+                config = "get=oldest";
+                break;
+            case 2:
+                config = "get=stable";
+                break;
+            case 3:
+                config = "get=all_durable";
+                break;
+            case 4:
+            default:
+                return;
+            }
+
+            std::string hex_ts;
+            bool ts_supported;
+            int ret = conn->query_timestamp(config, hex_ts, ts_supported);
+
+            if (ret != 0)
+                throw "'query_timestamp' failed with ret value: '" + std::to_string(ret) +
+                  "', and config: '" + config + "'";
+
+            if (hex_ts == "0")
+                hex_ts = "not specified";
+
+            print_border_msg("- Timestamp queried - " + config + ": " + hex_ts, GREEN);
+        } catch (const std::string &exception_str) {
+            print_border_msg("exception: " + exception_str, RED);
+        }
+
+    } while (true);
+}
+
+void
+interface_begin_transaction(session_simulator *session)
+{
+    std::string config = "";
+    std::vector<std::string> options;
+    options.push_back("Read timestamp");
+    options.push_back("Roundup read timestamps");
+    options.push_back("Roundup prepare timestamps");
+    options.push_back("Run operation");
+    options.push_back("<- go back");
+
+    do {
+        try {
+            print_options(options);
+
+            int choice = choose_num(1, options.size(), "Begin transaction >>");
+
+            switch (choice) {
+            case 1:
+                config += "read_timestamp=";
+                config += get_input("Enter read timestamp (in hex): ");
+                break;
+            case 2:
+                config += "roundup_timestamps=(read=true)";
+                break;
+            case 3:
+                config += "roundup_timestamps=(prepare=true)";
+                break;
+            case 4: {
+                int ret = session->begin_transaction(config);
+
+                if (ret != 0)
+                    throw "begin_transaction failed with return value: " + std::to_string(ret);
```

Comment:
I think we should go back to the main menu in this case.

Reply:
You could just return where you catch the error.

Reply:
Agreed! Just pushed a commit to do the same. [30d0e9d](https://github.com/wiredtiger/wiredtiger/pull/8417/commits/30d0e9d60cc7c0a517d4c38444c64374f54e269a)

Reply:
Even if it runs successfully better is to go to the main menu, cause there's nothing else you could do.

Reply:
That too!